### PR TITLE
Change server version error to exception rather that system stop call

### LINKED ISSUE #408: Operator start behavior
The Operator.start method is implemented like:

```java
public void start() {
   ...
    try {
      ...
    } catch (Exception e) {
      log.error("Error retrieving the server version. Exiting!", e);
      System.exit(1);
    }
  }
```

I wonder if instead of doing a `System.exit(1)`, the start method should thrown an exception and let the caller decide what to do.

### PR DESCRIPTION
**Pull request by teszes**
One way to close #408

### CODE REVIEW

File: operator-framework-core/src/main/java/io/javaoperatorsdk/operator/Operator.java [hunk 1]
```diff
@@ -51,7 +51,7 @@ public void start() {
       }
     } catch (Exception e) {
       log.error("Error retrieving the server version. Exiting!", e);
-      System.exit(1);
+      throw new OperatorException("Error retrieving the server version");
```
Comment (attilameszaros-form3) [inline operator-framework-core/src/main/java/io/javaoperatorsdk/operator/Operator.java:54]: We should add the root cause exception, are we sure that this is always version?

Comment (metacosm) [reply operator-framework-core/src/main/java/io/javaoperatorsdk/operator/Operator.java:54]: It most often is, but, strictly speaking, there might be other causes.

Comment (teszes) [reply operator-framework-core/src/main/java/io/javaoperatorsdk/operator/Operator.java]: chained the thrown exception to the new one, would this work the way you expect?

**Comment by metacosm** [comment]
Have you checked that throwing the exception properly crashes the app, resulting in a new pod being brought up when running in a cluster environment?

**Review by metacosm — APPROVED** [review]

**Pull request merged**